src/PathOrderMonotonic.h

```diff
@@ -0,0 +1,449 @@
+    size_t getFarthestEndpoint(Path* polyline, const Point point)
+    {
+        const coord_t front_dist = vSize2(polyline->converted->front() - point);
+        const coord_t back_dist = vSize2(polyline->converted->back() - point);
+        if(front_dist < back_dist)
+        {
+            return polyline->converted->size() - 1;
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    /*!
+     * Find which lines are overlapping with a certain line.
+     * \param polyline_it The line with which to find overlaps. Given as an
+     * iterator into the sorted polylines list, to cut down on the search space.
+     * If the lines don't have too much overlap, this should result in only a
+     * handful of lines being searched at all.
+     * \param perpendicular A vector perpendicular to the monotonic vector, pre-
+     * calculated.
+     * \param polylines The sorted list of polylines.
+     */
+    std::vector<Path*> getOverlappingLines(const typename std::vector<Path*>::iterator polyline_it, const Point perpendicular, const std::vector<Path*>& polylines)
+    {
+        //How far this extends in the monotonic direction, to make sure we only go up to max_adjacent_distance in that direction.
+        const coord_t start_projection = dot((*polyline_it)->converted->front(), monotonic_vector);
+        const coord_t end_projection = dot((*polyline_it)->converted->back(), monotonic_vector);
+        const coord_t my_farthest_projection = std::max(start_projection, end_projection);
+        const coord_t my_closest_projection = std::min(start_projection, end_projection);
+        //How far this line reaches in the perpendicular direction -- the range at which the line overlaps other lines.
+        coord_t my_start = dot((*polyline_it)->converted->front(), perpendicular);
+        coord_t my_end = dot((*polyline_it)->converted->back(), perpendicular);
+        if(my_start > my_end)
+        {
+            std::swap(my_start, my_end);
+        }
+
+        std::vector<Path*> overlapping_lines;
+        for(auto overlapping_line = polyline_it + 1; overlapping_line != polylines.end(); overlapping_line++)
+        {
+            //Don't go beyond the maximum adjacent distance.
+            const coord_t start_their_projection = dot((*overlapping_line)->converted->front(), monotonic_vector);
+            const coord_t end_their_projection = dot((*overlapping_line)->converted->back(), monotonic_vector);
+            const coord_t their_farthest_projection = std::max(start_their_projection, end_their_projection);
+            const coord_t their_closest_projection = std::min(start_their_projection, end_their_projection);
+            if(their_closest_projection - my_farthest_projection > max_adjacent_distance * 1000
+                    || my_closest_projection - their_farthest_projection > max_adjacent_distance * 1000)  //Distances are multiplied by 1000 since monotonic_vector is not a unit vector, but a vector of length 1000.
```

Comment:
Either make (global) `constexpr`, or use `MM2INT(...)`.

Reply:
Made it a static constexpr here: a0656e670f3470197d3def1612186ec759478ebb